ch_backup/clickhouse/control.py

```diff
@@ -284,6 +284,17 @@
 """
 )
 
+GET_SHARD_AND_REPLICA_NAME = strip_query(
+    """
+    SELECT
+        last_value_respect_nullsArray(splitByString('/', zookeeper_path)) AS shard_name,
+        last_value_respect_nullsArray(splitByString('/', replica_path)) AS replica_name
```

Comment:
replica_name is also directly available in the table, but during tests is was always null. Thus I opted for splitting the replica_path...What it is better?

Reply:
I'm not sure in this method. `zookeeper_path` can be the same for all shards and replicas. How about reading from `system.macros` ? As a rule it contains appropriate values for each replica. But this is not very reliable method.

@Alex-Burmak what do you think ?

Reply:
Agree. `system.replicas` is a wrong place for getting shard and replica names. It contains per table information. `zookeeper_path` and `replica_path` can be almost arbitrary values.

Vote for `system.macros`.

Reply:
Can we use `replica_name` column directly from `system.replicas` instead of splitting `replica_path`? [docs](https://clickhouse.com/docs/ru/operations/system-tables/replicas)

Reply:
thanks @Alex-Burmak. What would be best? `system.macros` or outside CH (see this [comment](https://github.com/yandex/ch-backup/pull/87/files#r1389337953))? what would a reliable way without CH to get this information?